src/textractComprehendMedical.ts

```diff
@@ -0,0 +1,125 @@
+import * as path from 'path';
+import { Duration } from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+
+export interface TextractComprehendMedicalProps {
+  /** memory of Lambda function (may need to increase for larger documents) */
+  readonly lambdaMemoryMB?: number;
+  readonly lambdaTimeout?: number;
+  readonly textractComprehendMedicalFunction?: lambda.IFunction;
+  readonly lambdaLogLevel?: string;
+  readonly s3InputBucket?: string;
+  readonly comprehendMedicalRoleName?: string;
+  readonly comprehendMedicalJobType?: string;
+  /** prefix for the incoming document. Will be used to create role */
+  readonly s3InputPrefix?: string;
+  /** List of PolicyStatements to attach to the Lambda function for S3 GET and LIST. */
+  readonly inputPolicyStatements?: iam.PolicyStatement[];
+}
+
+/**
+ * This construct takes in a manifest definition or a plain JSON with a s3Path:
+ *
+ * example s3Path:
+ * {"s3Path": "s3://bucketname/prefix/image.png"}
+ *
+ *
+ * Then it generated the numberOfPages attribute and the mime on the context.
+ * The mime types checked against the supported mime types for Textract and if fails, will raise an Exception failing the workflow.
+ *
+ * Example (Python)
+ * ```python
+ decider_task_id = tcdk.TextractPOCDecider(
+ self,
+ f"InsuranceDecider",
+ )
+ ```
+
+ *
+ */
+export class TextractComprehendMedical extends sfn.StateMachineFragment {
+  public readonly startState: sfn.State;
+  public readonly endStates: sfn.INextable[];
+  public readonly textractComprehendMedicalFunction: lambda.IFunction;
+
+  constructor(parent: Construct, id: string, props: TextractComprehendMedicalProps) {
+    super(parent, id);
+
+    var lambdaMemoryMB = props.lambdaMemoryMB === undefined ? 1024 : props.lambdaMemoryMB;
+    var lambdaTimeout = props.lambdaTimeout === undefined ? 900 : props.lambdaTimeout;
+    var lambdaLogLevel = props.lambdaLogLevel === undefined ? 'INFO' : props.lambdaLogLevel;
+    var s3InputPrefix = props.s3InputPrefix === undefined ? 'uploads' : props.s3InputPrefix;
+    var cmJobType = props.comprehendMedicalJobType === undefined ? 'ICD10' : props.comprehendMedicalJobType;
+
+    const comprehendMedicalRole = new iam.Role(this, 'RoleComprehendMedical', {
+      assumedBy: new iam.ServicePrincipal('comprehendmedical.amazonaws.com'),
+    });
+    comprehendMedicalRole.addToPolicy(
+      new iam.PolicyStatement({
+        actions: ['s3:GetObject', 's3:ListBucket', 's3:PutObject'],
+        effect: iam.Effect.ALLOW,
+        resources: ['*'],
```

Comment:
Any chance we can narrow this down?

Reply:
I changed the code to cover both situations 1/ when I have input bucket and 2/ when I don't have input bucket